gtsfm/averaging/rotation/shonan.py

```diff
@@ -50,31 +74,44 @@ def __init__(self, two_view_rotation_sigma: float = _DEFAULT_TWO_VIEW_ROTATION_S
         self._p_min = 5
         self._p_max = 64
 
+        # Build robust error function.
+        self._robust_error_model = robust_error_model
+        if self._robust_error_model is not None:
+            self._robust_error_model = getattr(gtsam.noiseModel.mEstimator, robust_error_model.value)
+            if robust_error_model_kwargs is None:
+                robust_error_model_kwargs = DEFAULT_ROBUST_MODEL_KWARGS[robust_error_model]
+            self._robust_error_model = self._robust_error_model(**robust_error_model_kwargs)
+            self._p_max = self._p_min
+
+
     def __get_shonan_params(self) -> ShonanAveragingParameters3:
         lm_params = LevenbergMarquardtParams.CeresDefaults()
         shonan_params = ShonanAveragingParameters3(lm_params)
-        shonan_params.setUseHuber(False)
-        shonan_params.setCertifyOptimality(True)
+        shonan_params.setUseHuber(self._robust_error_model is not None)
+        shonan_params.setCertifyOptimality(self._robust_error_model is None)
         return shonan_params
 
     def __between_factors_from_2view_relative_rotations(
-        self, i2Ri1_dict: Dict[Tuple[int, int], Rot3], old_to_new_idxs: Dict[int, int]
-    ) -> BetweenFactorPose3s:
+        self, 
+        i2Ri1_dict: Dict[Tuple[int, int], Rot3],
+        corr_idxs: Dict[Tuple[int, int], np.ndarray],
+        old_to_new_idxs: Dict[int, int],
+        use_robust_error: bool,
+    ) -> gtsam.BinaryMeasurementsRot3:
         """Create between factors from relative rotations computed by the 2-view estimator."""
         # TODO: how to weight the noise model on relative rotations compared to priors?
-        noise_model = gtsam.noiseModel.Isotropic.Sigma(POSE3_DOF, self._two_view_rotation_sigma)
-
-        between_factors = BetweenFactorPose3s()
-
+        measurements = gtsam.BinaryMeasurementsRot3()
         for (i1, i2), i2Ri1 in i2Ri1_dict.items():
-            if i2Ri1 is not None:
+            if i2Ri1 is not None and len(corr_idxs[(i1, i2)]) > 0:
                 # ignore translation during rotation averaging
-                i2Ti1 = Pose3(i2Ri1, np.zeros(3))
+                noise_model = gtsam.noiseModel.Isotropic.Sigma(ROT3_DOF, 1 / corr_idxs[(i1, i2)].shape[0])
```

Comment:
@travisdriver can you add a flag that allows us to run without using inliers as uncertainty? e.g. to support an ablation as we have it currently, or in case it can't converge with the uncertainty-weighted cost